This is a bug in the upstream code. They reset the vector to 0, then try to access a value in a helper function
It only shows up in our builds since we build with _GLIBCXX_ASSERTIONS
--- a/dcgmlib/src/DcgmTopology.cpp
+++ b/dcgmlib/src/DcgmTopology.cpp
@@ -782,7 +782,7 @@ dcgmReturn_t GetActiveNvSwitchNvLinkCountsForAllGpus(const std::vector<dcgm_topo
         return DCGM_ST_INSUFFICIENT_SIZE;
     }
 
-    gpuCounts = {};
+    gpuCounts.assign(gpuInfo.size(), 0);
 
     // Attempt to query through NVML. This is not supported in all drivers so
     // there is a fallback approach below

